### ISSUE TITLE
How to test on free size of images?

### ISSUE THREAD

**Issue by TrinhQuocNguyen** [MathiasGruber/PConv-Keras#4]
Hi MathiasGruber,
(and Hello guys)
At the training step, you have used 512x512 pixels images.
How can we test free size images ( lets say 640x640 pixels images)?

Thank you.

**Comment by MathiasGruber** [comment]
I implemented a possible solution to this in a new notebook. It should be possible to setup the network to train on arbitrary image sizes as well, however, memory usage quite quickly explodes. I'd love to hear other ideas on how to approach this for large images without requiring a huge amount of memory.

**Comment by KhyatiGanatra** [comment]
Hello, this works nicely for the sample image given. My goal is to provide my own masked image and mask. I do not want to generate random masks. I Would you suggest if I should change the way random_masks is defined or directly pass the mask in the predict function?

**Comment by TrinhQuocNguyen** [comment]
> I implemented a possible solution to this in a new notebook. It should be possible to setup the network to train on arbitrary image sizes as well, however, memory usage quite quickly explodes. I'd love to hear other ideas on how to approach this for large images without requiring a huge amount of memory.

Hi MathiasGruber, could you please make a demo how to test on arbitrary image even it quickly explodes the memory? 👍 🤓

**Comment by TrinhQuocNguyen** [comment]
> Hello, this works nicely for the sample image given. My goal is to provide my own masked image and mask. I do not want to generate random masks. I Would you suggest if I should change the way random_masks is defined or directly pass the mask in the predict function?

Hi KhyatiGanatra, you can do that by making a function to load your own masks and replace "random_masks" function by it in test phase

**Comment by cpitman** [comment]
@MathiasGruber Haven't gotten to this point yet, but I don't think there is anything in this network preventing you from training on smaller images but then using the trained weights on an arbitrary sized image. Since all the stages are convolutions, the number of parameters stays the same.

**Comment by MathiasGruber** [comment]
> @MathiasGruber Haven't gotten to this point yet, but I don't think there is anything in this network preventing you from training on smaller images but then using the trained weights on an arbitrary sized image. Since all the stages are convolutions, the number of parameters stays the same.

That should work, but as soon as the image size increase memory explodes. The best way I found so far is the image chunking approach, but it'd be cool if anyone can find a better approach.
